torchaudio/csrc/transducer.cpp

```diff
@@ -17,11 +17,38 @@ int64_t cpu_rnnt_loss(
     torch::Tensor grads,
     int64_t blank_label,
     int64_t num_threads) {
+  TORCH_CHECK(labels.dtype() == torch::kInt32, "labels must be int32 type");
+  TORCH_CHECK(
+      label_lengths.dtype() == torch::kInt32,
+      "label_lengths must be int32 type");
+  TORCH_CHECK(
+      input_lengths.dtype() == torch::kInt32, "lengths must be int32 type");
+  TORCH_CHECK(acts.is_contiguous(), "log_probs must be contiguous");
+  TORCH_CHECK(labels.is_contiguous(), "labels must be contiguous");
+  TORCH_CHECK(
+      label_lengths.is_contiguous(), "label_lengths must be contiguous");
+  TORCH_CHECK(input_lengths.is_contiguous(), "lengths must be contiguous");
+  TORCH_CHECK(
+      input_lengths.size(0) == acts.size(0), "must have a length per example.");
+  TORCH_CHECK(
+      label_lengths.size(0) == acts.size(0),
+      "must have a label length per example.");
+  TORCH_CHECK(acts.dim() == 4, "log_probs must be 4D");
```

Comment:
I am aware that these error messages are carried out from the original code, but I do not think `must be 4D` is user friendly. If the function is expecting the fixed shape, it should communicate that.  `4D (foo, bar, baz, boo)`.